packages/eslint-plugin-office-addins/test/rules/call-sync-before-read.test.ts

```diff
@@ -0,0 +1,121 @@
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import rule from '../../src/rules/call-sync-before-read';
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('call-sync-before-read', rule, {
+  valid: [ 
+    {
+      code: `
+      Excel.run(function (context) {
+        var selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('address');
+        await context.sync()
+        console.log('The selected range is: ' + selectedRange.address);
+      });`
+    },
+    {
+      code: `
+      Excel.run(function (context) {
+        var selectedRange;
+        selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('address');
+        await context.sync()
+        console.log('The selected range is: ' + selectedRange.address);
+      });`
+    },
+    {
+      code: `
+      Excel.run(function (context) {
+        var selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('address');
+        return context.sync()
+          .then(function () {
+            console.log('The selected range is: ' + selectedRange.address);
+        });
+      })`
+    },
+    {
+      code: `
+      Excel.run(function (context) {
+        var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+        await context.sync();
+        if (dataSheet.isNullObject) {
+          dataSheet.position = 1;
+        }
+      })`
+    },
+    {
+      code: `
+      Excel.run(function (context) {
+        var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+        if (true) {
+          await context.sync();
+          dataSheet.position = 1;
+        }
+      })`
+    },
+  ],
+  invalid: [
+    {
+      code: `
+      Excel.run(function (context) {
+        var selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('address');
+        console.log('The selected range is: ' + selectedRange.address);
+      });`,
+      errors: [{ messageId: "callSync", data: { name: "selectedRange" } }]
+    },
+    {
+      code: `
+      Excel.run(function (context) {
+        var selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('address');
+        console.log('The selected range is: ' + selectedRange.address);
+        await context.sync();
+      });`,
+      errors: [{ messageId: "callSync", data: { name: "selectedRange" } }]
+    },
+    {
+      code: `
```

Comment:
Could you make a test where `selectedRange` loads the `address`, there is a `context.sync`, but `selectedRange2` is the one that makes the read operation? This is to check if the code is handling different variables at the same time.

Reply:
This rule doesn't check the loads . . . just the reads of the original variable.  I think the load being called before being used is part of other rules and maybe this cases you mention here (needing to check the sync) is the P1.5 rule we have yet to start.